core/src/main/java/org/mule/DefaultMessageCollection.java

```diff
@@ -274,7 +274,9 @@ public int size()
         }
         else
         {
-            throw new UnsupportedOperationException("getPayloadAsBytes(), use getPayload(DataType.BYTE_ARRAY_DATA_TYPE)");
+            throw new UnsupportedOperationException(
+                    "getPayloadAsBytes(), There has been an attempt to directly access the payload of a message collection, which is unsupported. " + System.getProperty("line.separator") +
+                                                    "Please retrieve the value from messageList or use getPayload(DataType.BYTE_ARRAY_DATA_TYPE)");
```

Comment:
According to [SE-3746] (https://www.mulesoft.org/jira/browse/SE-3746) the new exception description was accepted.